### ISSUE TITLE
Didn't manage to get it to work on another glacier yet

### ISSUE THREAD

**Issue by fmaussion** [OGGM/oggm-3dviz#7]
Here is a dataset from another, larger glacier (Aletsch): https://cluster.klima.uni-bremen.de/~fmaussion/share/oggm/aletsch.nc

The format is closer to what OGGM outputs normally, and I called the tool with:

```python
import xarray as xr
from tools import Glacier3DViz

f = 'aletsch.nc'

ds_glacier = xr.open_dataset(f)
ds_glacier = ds_glacier.isel(x=slice(70, 180), y=slice(50, 250)).load()

viz = Glacier3DViz(
    ds_glacier,
    "distributed_thickness",
    topo_bedrock = 'bedrock'
)

viz.set_topo_texture(use_cache=False)
```

There is an issue with the download of the satellite data:

```
/home/mowglie/.miniconda3/envs/oggm-3dviz/lib/python3.11/site-packages/pystac_client/client.py:187: NoConformsTo: Server does not advertise any conformance classes.
  warnings.warn(NoConformsTo())
/home/mowglie/.miniconda3/envs/oggm-3dviz/lib/python3.11/site-packages/stackstac/prepare.py:364: UserWarning: The argument 'infer_datetime_format' is deprecated and will be removed in a future version. A strict version of it is now the default, see https://pandas.pydata.org/pdeps/0004-consistent-to-datetime-parsing.html. You can safely remove this argument.
  times = pd.to_datetime(
Traceback (most recent call last):
  File "/home/mowglie/disk/Dropbox/HomeDocs/git/oggm-3dviz/anim.py", line 17, in <module>
    viz.set_topo_texture(use_cache=False)
  File "/home/mowglie/disk/Dropbox/HomeDocs/git/oggm-3dviz/tools/viz.py", line 43, in set_topo_texture
    self.topo_texture = get_topo_texture(bbox, srs=srs, use_cache=use_cache)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/mowglie/disk/Dropbox/HomeDocs/git/oggm-3dviz/tools/texture.py", line 149, in get_topo_texture
    da_img_raw.to_zarr(temp_zarr_dataset)
  File "/home/mowglie/.miniconda3/envs/oggm-3dviz/lib/python3.11/site-packages/xarray/core/dataarray.py", line 4215, in to_zarr
    return to_zarr(  # type: ignore[call-overload,misc]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/mowglie/.miniconda3/envs/oggm-3dviz/lib/python3.11/site-packages/xarray/backends/api.py", line 1649, in to_zarr
    zstore = backends.ZarrStore.open_group(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/mowglie/.miniconda3/envs/oggm-3dviz/lib/python3.11/site-packages/xarray/backends/zarr.py", line 454, in open_group
    zarr_group = zarr.open_group(store, **open_kwargs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/mowglie/.miniconda3/envs/oggm-3dviz/lib/python3.11/site-packages/zarr/hierarchy.py", line 1547, in open_group
    raise ContainsGroupError(path)
zarr.errors.ContainsGroupError: path '' contains a group

Process finished with exit code 1
```

I'll discuss with @pat-schmitt first.

**Comment by pat-schmitt** [comment]
For me, it is working (see attached video).

But there is a problem with `'bedrock`, it looks like the bedrock is 0 outside of the glacier everywhere. I think this is the reason why in the video the satellite image is only visible where the glacier has already retreated.

https://github.com/OGGM/oggm-3dviz/assets/50843444/6f7bc208-da26-4d0b-884d-107f3882f665

**Comment by fmaussion** [comment]
Ohhh, my bad! Will check again

**Comment by fmaussion** [comment]
OK the file had an issue, updated now:  https://cluster.klima.uni-bremen.de/~fmaussion/share/oggm/aletsch.nc

**Comment by fmaussion** [comment]
After some testing, it looks like the crs is not well understood by the satellite download library. We will work on this.

**Comment by pat-schmitt** [comment]
Has this worked for you without converting the original data into UTM? Because I needed to make some adaptations to work for me with a non-UTM coordinate system (see PR https://github.com/OGGM/oggm-3dviz/pull/4).

**Comment by benbovy** [comment]
> Has this worked for you without converting the original data into UTM?

I don't know how this could have worked, actually. Maybe because I didn't clear my local cache for the downloaded sentinel data? That's weird, though.

I just re-tried downloading the glacier data from https://github.com/OGGM/oggm-3dviz/issues/7#issuecomment-1691335401 and I see that the CRS seems to be slightly different than the one of the downloaded sentinel data (i.e., Transverse Mercator with custom longitude origin vs. UTM zone 32N)?

I think your changes in #4 should work but it still fails (some 0-sized array later in the processing) because of the projection mismatch between the sentinel data vs. the last aletsch data that I downloaded. Maybe we could implement automatic re-projection of the glacier data (coarser resolution)? We should at least check that the two projections match and raise a meaningful error if not.

**Comment by pat-schmitt** [comment]
Yes I also had the 0-sized array problem, but I could solve it by including the changes here https://github.com/OGGM/oggm-3dviz/pull/4/files#diff-85af32bfe80002340920a152470dbdeb10b7858df6bdd308467248b96ede0c08R51, so I reprojected `bbox_lonlat` back to the crs of the satellite data. This was working previously when using UTM because you used `bbox` instead of `bbox_lonlat` previously here https://github.com/OGGM/oggm-3dviz/pull/4/files#diff-85af32bfe80002340920a152470dbdeb10b7858df6bdd308467248b96ede0c08R59, and if `bbox` is in UTM it is fine, but if not this leads to an 0-sized array. Or is your problem coming from somewhere else?

But indeed it is probably easier to re-project the data to UTM before we do the downloading and selection of the sentinel data. Than we can simplify the downloading routine a lot...

**Comment by benbovy** [comment]
Nevermind, I cleared the cache and re-run the example with your changes (#4) and it worked! Sorry for the confusion.

I haven't looked much into how exactly works `texture` and `texture_map_to_plane` in pyvista, but I guess that just having a matching bbox worked here because the two projections (glacier data vs. sentinel imagery) are rather similar. It's probably not enough in the cases where the projections differ more, though.

> But indeed it is probably easier to re-project the data to UTM before we do the downloading and selection of the sentinel data. Than we can simplify the downloading routine a lot...

Yes, and more generally downloading and processing GBs of raw sentinel data just for getting the terrain texture is a bit overkill. It would be great to have another tiled data source (STAC API, WMS, etc.) with pre-processed (true-color) terrain texture.

**Comment by benbovy** [comment]
I think that all the issues mentioned here should now be all resolved with #12 merged?

**Comment by pat-schmitt** [comment]
Yes, I also think so
